### DISCUSSION TITLE
When using live server, ctrl + click on any internal link does not open in new tab

### DISCUSSION THREAD

**Discussion by tars0x9752** [srid/emanote#383] [Q&A]
**Describe the bug**

When using live server, `ctrl + click` on any internal link does not open in new tab. I think this is caused by ema instead of emanote, but I'm posting this here because I've faced this problem while using emanote.

I guess the click event is intercepted by the embedded JS code, which is probably where "`ctrl + click` to open in new tab" is disabled.

BTW I'm not having that much trouble, since it's only when using live server and I'm just reporting this in case it was a bug. Sorry if it's intended and not a bug.

**To Reproduce**
1. Launch the live server.
2. Open any page that has an internal link.
3. `ctrl + click` on an internal link.
4. New tab won't open. The link opens in the same tab instead.

**Expected behavior**

`ctrl + click` on a link will automatically open it in a new tab.

**Desktop (please complete the following information):**
 - Browser: Firefox
 - Version: 107.0 (64-bit)

**Comment by srid** [comment]
I cannot reproduce this on macOS (where it is <kbd>Cmd+Click</kbd>), both on Brave and Firefox.

Either your issue is Linux or Windows only (can somebody else confirm?), or something unique to your environment (browser extensions getting in the way?).

**Reply by tars0x9752** [reply]
I use x86_64 Linux (NixOS, X, i3wm). This behavior was reproduced without any browser extension. Also reproduced with google chrome (as it is just after installation). Not sure how it works on Windows. I have no clue why it doesn't reproduce on macOS though.

One thing to note: Middle click and their equivalents do not reproduce this and just work as expected. (FWIW)

**Reply by srid** [reply]
I'd file an issue on the Ema repo instead. Here's the code that handles link clicks: https://github.com/EmaApps/ema/blob/61faae56aa0f3c6ca815f344684cc566f6341662/ema/www/ema-shim.js#L102-L121

**Reply by tars0x9752** [reply]
After looking at the code for a bit, It doesn't seem strange that `ctrl+click` doesn't work with that code. Because it doesn't explicitly handle ctrl key press of the click event while invoking `preventDefault` on the click event. (IIRC,  `preventDefault` literally prevents the native `opens new tab` behavior.)

However I'm not sure why no one but myself has been able to reproduce it. :thinking:

**Reply by jfpedroza** [reply]
It's happening to me too. I'm on EndeavourOS, i3 and Firefox. Middle click works as expected.

**Comment by precompute** [comment]
Can't reproduce this either.  Debian Stable + XFCE4, FF / Brave

**Reply by tars0x9752** [reply]
Hmm, am I the only one with this problem? Then, Idk but It could be a problem with my environment. :/
